src/lib/expansion/expansion-panel.ts

```diff
@@ -127,12 +147,31 @@ export class MatExpansionPanel extends _MatExpansionPanelMixinBase
     return this.expanded ? 'expanded' : 'collapsed';
   }
 
+  ngAfterViewInit() {
+    this._portal = new TemplatePortal<any>(this._content, this._viewContainerRef);
+
+    // Render the content as soon as the panel becomes open.
+    this.opened.pipe(startWith(null!)).subscribe(() => {
+      if (this.expanded && !this._outlet.hasAttached()) {
+        this._outlet.attach(this._portal);
+      }
+    });
+
+    // Remove the content once the panel is closed and the closing animation is done.
```

Comment:
What is the reasoning behind removing the content on closing?  Since the issue we are looking to address is lazy loading, is there a reason to detach and reattach all the time?

Reply:
I was a bit on the fence on this one as well, but it could help if you have some really massive panels with lots of elements. I'm open to not doing this though.

Reply:
Could this be configured perhaps via an Input?
I'm developing (for learning purposes) an RSS feed reader. On desktop, each item is placed inside an expansion panel (for fancyness (?)); inspired by Google Inbox interface.

Removing the content inside the panel after been read, could be something I'd like to happen. Being that as more and more items are been loaded, more resources will be consumed. (O course I shall do something about this regardless of this specific feature).

However, also thinking about other Google products like GSuit admin console: where some settings are loaded asynchronously the first time, but then kept. (I don't know if they're actually removed from DOM, but I wouldn't remove them in such scenario).

Reply:
I think it might be best at this point to not remove the content once it is placed and we can determine if it is something that we want to allow later, potentially as @bardiel mentioned via `@Input`